### PR TITLE
Microsoft headers phishing

### PR DESCRIPTION
**Pull request by tomer-pan**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34329

## Description
Adding a check for Microsoft's email headers in order to get PCL/SCL/BCL scores.

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Packs/Phishing/ReleaseNotes/2_2_7.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+#### Playbooks
+##### Phishing Investigation - Generic v2
+- Use the "Process Microsoft's Anti-Spam Headers" sub-playbook in order to assess PCL/SCL/BCL scores.
+
+##### New: Process Microsoft's Anti-Spam Headers
+- This playbook will return SCL, BCL and PCL scores to the context (if exist).
+It will also do the following:
+1) Set the email classification to "spam" if the "SCL" score is equal or higher than 5.
+2) Set a value of "Medium" to a "MicrosoftHeadersSeverityCheck" field in the context if the PCL or BCL value is higher than 4.
```
Comment (David-BMS) [inline Packs/Phishing/ReleaseNotes/2_2_7.md:10]: @tomer-pan This should be hardcoded? maybe worth to add a playbook input to define which number is equal to each value

Comment (tomer-pan) [reply Packs/Phishing/ReleaseNotes/2_2_7.md:10]: @David-BMS  These numbers are based on Microsoft's scoring as mentioned here: 
SCL - https://docs.microsoft.com/en-us/microsoft-365/security/office-365-security/spam-confidence-levels?view=o365-worldwide
BCL - https://docs.microsoft.com/en-us/microsoft-365/security/office-365-security/bulk-complaint-level-values?view=o365-worldwide
PCL - https://docs.microsoft.com/en-us/Exchange/antispam-and-antimalware/antispam-protection/antispam-stamps?view=exchserver-2019

**Comment by tomer-pan** [comment]
@richardbluestone @yaron-libman In this PR, I've added a new sub-playbook for processing Microsoft headers (with a full description of which values are being processed). It was added to the "Phishing Investigation - Generic v2" playbook and also, the "Calculate Severity - Generic v2" playbook was changed in order to calculate the severity based on the values on these headers as well. Please add it to the Phishing docs you are currently working on. If you have any questions, please don't hesitate to reach out!

File: Packs/Phishing/ReleaseNotes/2_2_7.md (hunk 1 above)
Comment (tomer-pan) [reply Packs/Phishing/ReleaseNotes/2_2_7.md]: @David-BMS As we discussed, I added playbook inputs for setting a severity for each value.

File: Packs/Phishing/IncidentFields/incidentfield-bclscore.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "associatedTypes": [
+        "Phishing"
+    ],
+    "associatedToAll": false,
+    "unmapped": false,
+    "unsearchable": false,
+    "caseInsensitive": true,
+    "sla": 0,
+    "threshold": 72,
+    "fromVersion": "5.5.0"
```
Comment (David-BMS) [inline Packs/Phishing/IncidentFields/incidentfield-bclscore.json:30]: @tomer-pan New content will be 6.0 please review for all files

**Comment by richardbluestone** [comment]
Update the docs. You will need to re-generate the readmes (for the playbooks).

File: Packs/Phishing/Playbooks/Process_Microsoft's_Anti-Spam_Headers.yml [hunk 3]
```diff
@@ -0,0 +1,1379 @@
...
+  description: |
+    Possible Values:
+
+    Medium: PCL or BCL scores are equal to or higher than 4.
+
+    High: BCL score is equal to or higher than 8.
+  type: unknown
+tests:
+- No tests (auto formatted)
+fromversion: 5.0.0
```
Comment (David-BMS) [inline Packs/Phishing/Playbooks/Process_Microsoft's_Anti-Spam_Headers.yml:1379]: @tomer-pan still 5.0

**Review by David-BMS — APPROVED** [review]

**Pull request merged**